File: Server/Server/Models/View_Models/TaskModel.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Server.Models.View_Models
{
    public class TaskModel
    {
        public int TaskId { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Description { get; set; }

        [Required]
        public string StartTask { get; set; }

        [Required]
        public string Periodicity { get; set; }

        [Required]
        public string FilterText { get; set; }

        [Required]
        public int ApiId { get; set; }
    }
}

File: Client/src/app/models/taskModel.ts
export class TaskModel {

  constructor(name: string, description: string, startTask: string, periodicity: string, filterText: string, apiId: number,
              taskId: number, lastExecution: string, userId: number) {
    this.name = name;
    this.description = description;
    this.startTask = startTask;
    this.periodicity = periodicity;
    this.filterText = filterText;
    this.apiId = apiId;
    this.taskId = taskId;
    this.lastExecution = lastExecution;
    this.userId = userId;
  }

  taskId: number;
  lastExecution: string;
  userId: number;
  name: string;
  description: string;
  startTask: string;
  periodicity: string;
  filterText: string;
  apiId: number;
}
